platform/ui/src/components/Button/Button.jsx

```diff
@@ -26,60 +30,99 @@ const disabledClasses = {
+    white:
+      'text-white hover:bg-white hover:text-black active:opacity-80 focus:bg-white focus:text-black',
   },
   outlined: {
-    default: 'border border-black text-black hover:bg-gray-200',
-    primary: 'border border-blue-900 text-blue-900 hover:bg-blue-100',
-    secondary: 'border border-blue-300 text-blue-300 hover:bg-blue-100',
+    default:
+      'border bg-trasparent border-custom-aquaBright text-custom-aquaBright hover:opacity-80 active:opacity-100 focus:opacity-80',
+    primary:
+      'border bg-transparent border-custom-blue text-custom-blue hover:opacity-80 active:opacity-100 focus:opacity-80',
+    secondary:
+      'border bg-transparent border-custom-violetPale text-custom-violetPale hover:opacity-80 active:opacity-100 focus:opacity-80',
+    white:
+      'border bg-transparent border-white text-white hover:opacity-80 active:opacity-100 focus:opacity-80',
   },
   contained: {
-    default: 'bg-black hover:bg-gray-800 text-white',
-    primary: 'bg-blue-900 hover:bg-blue-800 text-white',
-    secondary: 'bg-blue-300 hover:bg-blue-500 text-black',
+    default:
+      'bg-custom-aquaBright text-white hover:opacity-80 active:opacity-100 focus:opacity-80',
+    primary:
+      'bg-custom-blue text-white hover:opacity-80 active:opacity-100 focus:opacity-80',
+    secondary:
+      'bg-custom-violetPale text-white hover:opacity-80 active:opacity-100 focus:opacity-80',
+    white:
+      'bg-white text-black hover:opacity-80 active:opacity-100 focus:opacity-80',
   },
 };
 
 const sizeClasses = {
-  small: 'py-1 px-2 text-sm',
-  medium: 'py-2 px-4 text-base',
-  large: 'py-3 px-6 text-lg',
+  small: 'py-2 px-2 text-base',
+  medium: 'py-2 px-2 text-lg',
+  large: 'py-2 px-6 text-xl',
+};
+
+const fullWidthClasses = {
+  true: 'flex w-full',
+  false: 'inline-flex',
 };
 
 const Button = ({
   children,
   variant = defaults.variant,
   color = defaults.color,
   size = defaults.size,
-  radius = defaults.radius,
+  rounded = defaults.rounded,
   disabled = defaults.disabled,
   type = defaults.type,
+  fullWidth = defaults.fullWidth,
   startIcon: startIconProp,
   endIcon: endIconProp,
   className,
   ...rest
 }) => {
-  const baseClasses =
-    'inline-flex items-center outline-none transition duration-300 ease-in-out font-bold focus:outline-none';
-
   const startIcon = startIconProp && (
-    <div className="mr-2">{startIconProp}</div>
+    <div className="mr-2">
+      {React.cloneElement(startIconProp, {
+        className: classnames('w-4 h-4 fill-current'),
+      })}
+    </div>
+  );
+
+  const endIcon = endIconProp && (
+    <div className="ml-2">
+      {React.cloneElement(endIconProp, {
+        className: classnames('w-4 h-4 fill-current'),
+      })}
+    </div>
   );
+  const buttonElement = useRef(null);
 
-  const endIcon = endIconProp && <div className="ml-2">{endIconProp}</div>;
+  const handleOnClick = e => {
+    buttonElement.current.blur();
```

Comment:
why are we dealing with Blur in a button?

Reply:
When we click on a Button, it gets focused, hence style rules for `:focus` are applied. This `blur` is just to avoid these styles to be applied when we click on the Button. Here's an example without calling `blur()`:

![Mar-05-2020 12-43-11](https://user-images.githubusercontent.com/1905961/75998400-904e7400-5edf-11ea-941d-e9f73855cf13.gif)

Open to ideas